src/block_ffm.rs

```diff
@@ -283,7 +285,7 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockFFM<L> {
                 fb.ffm_buffer.len() * (self.ffm_k * self.ffm_num_fields) as usize;
             if local_data_ffm_len < FFM_STACK_BUF_LEN {
                 // Fast-path - using on-stack data structures
-                let mut local_data_ffm_values: [f32; FFM_STACK_BUF_LEN as usize] =
+                let local_data_ffm_values: [f32; FFM_STACK_BUF_LEN as usize] =
```

Comment:
You sue this doesn't need to be mut?

Reply:
Was a bit surprised too, won't merge until end-to-end tested (works fine for training etc., that touches this part)

Reply:
@ggaspersic if you check the macro, it seems that a copy is being made there, am I seeing this right?
```
let mut local_data_ffm_values = $local_data_ffm_values;
```
implying it doesn't matter if input is mutable or not as it's copied anyways?

Reply:
@SkBlaz missed that, I think you are right yes. Would be less confusing if this was an always inlined method that took a mutable ref  as input.

Reply:
agreed, perhaps to be put on the todo for the blocks-specific endeavors